### PR TITLE
Execute methods and access properties of variables from Run Custom Code

### PR DESCRIPTION
**Pull request by melugi**
The previous behavior of Run Custom Code was that it would take in a string and then compile it using `System.Diagnostics.Process` and redirect the output of the compilation to the output variable. This PR adds a branch in the logic of Run Custom Code to allow for user to put in lines such as `{myDefinedString.Length}`, `{myFirstString.Concat(mySecondString)}`, and `{myObject.ClassMember()}`. These inputs are evaluated by a new function `InvokeTypeMember`. 

`InvokeTypeMember` takes the input of Run Custom Code split into four parts: 
1) The object which we're trying to invoke/access on
2) A string which is the name of the class member we want, 
3) The type of the object we're invoking/accessing on 
4) Finally a list of ScriptVariables from which to retrieve the arguments the user inputted. 

It then uses Reflections to retrieve the method info based off the input parameters and invokes it on the object.

Some issues with this implementation: 
* You can't run things like `String.Join` because String isn't a ScriptVariable. This needs a bit more research to fix, maybe even some larger changes, because there doesn't seem to be an easy way to get a fully qualified assembly name from `String`.
* You can only run a single command on a variable. Nothing like `myFirstString.Concat(mySecondString).ToUpper()`, but users can use multiple run custom code command to achieve the same goal. This could be fixed, but I don't think it's worth it given the alternative isn't that bad.

### CODE REVIEW

**Review by openbots-ff — CHANGES_REQUESTED** [review]
Could you add a sample script for this?

**Comment by melugi** [comment]
@fodfran Added the new sample and did some cleaning.

**Comment by openbots-ff** [comment]
> @fodfran Added the new sample and did some cleaning.

Please make the changes we discussed on teams

File: taskt/Core/Script.cs [hunk 1]
```diff
@@ -234,13 +233,16 @@ public string GetDisplayValue(string requiredProperty = "")
...
             {
-                DataTable dataTable = (DataTable)VariableValue;              
+                DataTable dataTable = (DataTable)VariableValue;
                 var dataRow = dataTable.Rows[CurrentPosition];
-                return Newtonsoft.Json.JsonConvert.SerializeObject(dataRow.ItemArray);            
+                return Newtonsoft.Json.JsonConvert.SerializeObject(dataRow.ItemArray);
+            }
+            else if(VariableValue.GetType() != typeof(string))
+            {
+                return VariableValue.ToString();
```
Comment (openbots-ff) [inline taskt/Core/Script.cs]: Won't this produce an error if you're returning a data type that isn't one of the ones above but also can't be converted into a string?

Comment (melugi) [reply taskt/Core/Script.cs]: `ToString` is defined in Object so if a specific object doesn't override it then it'll execute `Object.ToString` which by default returns the qualified class name of the object (i.e if you do it on a data row it'll return `System.Data.DataRow`).

While it isn't the most complete solution this change prevents exceptions from happening in ConvertToUserVariable such as in this line: https://github.com/CameronHerwig/taskt/blob/213f9fb7c0c2659f4c643d3d4344245421c17aca/taskt/Core/ExtensionMethods.cs#L201

and gives us a way to return string representations for values like Bools and Ints without causing an exception.

Comment (openbots-ff) [reply taskt/Core/Script.cs]: Okay then. Sounds good, thanks

**Review by openbots-ff — DISMISSED** [review]

**Review by CameronHerwig — APPROVED** [review]

**Review by openbots-ff — APPROVED** [review]

**Pull request merged**